src/test/com/Elessar/app/client/MyClientIntegTest.java

```diff
@@ -0,0 +1,233 @@
+            Assert.assertEquals(true, logonResp3.getSuccess());
+
+            LogonResponse logonResp4 = clientA.logOn("Shuai Hu", "4238902", clientA_Port);
+            Assert.assertEquals(true, logonResp4.getSuccess());
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(true, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            assert(false);
+        }
+    }
+
+
+    private void logOffTest() {
+        try {
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(true, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+            LogoffResponse logoffResp1 = clientA.logOff("Ertai Cai");
+            Assert.assertEquals(false, logoffResp1.getSuccess());
+
+            LogoffResponse logoffResp2 = clientA.logOff("Zhi Xu");
+            Assert.assertEquals(true, logoffResp2.getSuccess());
+
+            LogoffResponse logoffResp3 = clientA.logOff("Shuai Hu");
+            Assert.assertEquals(true, logoffResp3.getSuccess());
+
+            LogoffResponse logoffResp4 = clientA.logOff("Shuai Hu");
+            Assert.assertEquals(true, logoffResp4.getSuccess());
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(false, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            assert(false);
+        }
+    }
+
+    private void p2pMsgTest() {
+        try {
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.MESSAGES).find(or(eq(Message.FROM_USER, "Shuai Hu"),
+                                                                                                   eq(Message.FROM_USER, "Zhi Xu"))).iterator()) {
+                assertEquals(false, cursor.hasNext());
+            }
+
+            LogonResponse logonResp1 = clientA.logOn("Shuai Hu", "4238902", clientA_Port);;
+            assertEquals(true, logonResp1.getSuccess());
+
+            P2PMsgResponse p2PMsgResp1 = clientA.sendMessage("Shuai Hu", "Zhi Xu", "SB");
+            assertEquals(true, p2PMsgResp1.getSuccess());
+            assertEquals(false, p2PMsgResp1.getIsDelivered());
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.MESSAGES).find(eq(Message.FROM_USER, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
```

Comment:
get rid of this "if" and also verify the total number of documents returned are what we expect